### PR TITLE
docs: publish-task

### LINKED ISSUE #325: Document all public API's - Remove CS1591 NoWarn
All public API's should be documented :)

[build.log](https://github.com/user-attachments/files/17535441/build.log)

- [x] YamlStringEnumConverter - made internal
- [x] MultilineStringEmitter - should this be internal?
- [x] IDependsOn - pubic API 
- [x] SharplinerSerializer - made internal
- [x] TargetPathType
- [x] SharplinerConfiguration
- [x] TaskBuilderBase
- [x] VariablesReference #326
- [x] TemplateDefinitionCollection, TemplateDefinition, Template
- [x] PublicDefinitions
- [x] Pipeline, PipelineDefinition, PipelineDefinitionCollection
- [x] Parameter
- [x] VariableDefinition
- [x] Conditioned #334
- [x] ValidationsExtensions - public API
- [x] ISharplinerDefinitionCollection
- [x] IDefinitionValidation

tasks:
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\AzureDevOpsTask - #352
- [x] src/Sharpliner/AzureDevOps/Model/Tasks/Builders/ScriptTaskBuilder.cs - #350 
- [x] src/Sharpliner/AzureDevOps/Model/Tasks/ScriptTask.cs - #350 
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\BashTask - #358 
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Builders\BashTaskBuilder - #358
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Builders\CheckoutTaskBuilder - #357 
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\CheckoutTask - #357 
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\DownloadTask - #359
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Builders\DownloadTaskBuilder - #359
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Builders\PowerShellTaskBuilder - #360
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Builders\PwshTaskBuilder - #360
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\PowerShellTask - #360
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Marketplace\ArchiveFilesTask - #362
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Marketplace\CopyFilesTask - #362
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Marketplace\DeleteFilesTask - #362
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Marketplace\PublishCodeCoverageResultsTask - #362
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Marketplace\PublishTestResultsTask - #362
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Builders\DotNetTaskBuilder - #364
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Marketplace\DotNetBuildCoreCliTask - #364
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Marketplace\DotNetCoreCliTask - #364
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Marketplace\DotNetPackCoreCliTask - #364
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Marketplace\DotNetPublishCoreCliTask - #364
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Marketplace\DotNetPushCoreCliTask - #364
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Marketplace\DotNetRestoreCoreCliTask - #364
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Marketplace\DotNetTestCoreCliTask - #364
- [x] src\Sharpliner\AzureDevOps\Model\Tasks\Marketplace\UseDotNetTask - #364
- [ ] src\Sharpliner\AzureDevOps\Model\Tasks\PublishTask - #365 

~625 total warnings

### PR DESCRIPTION
**Pull request by Meir017**
closes #325

### CODE REVIEW

File: src/Sharpliner/AzureDevOps/AzureDevOpsDefinition.cs [hunk 1]
```diff
@@ -273,7 +273,32 @@ internal static LibraryReference<TDefinition> CreateLibraryRef<TLibrary, TDefini
...
+    ///   displayName: Publish artifact
+    ///   artifact: Binary
+    ///   continueOnError: false
+    /// </code>
+    /// <param name="artifactName">Artifact name</param>
+    /// <param name="filePath">Path to the folder or file you want to publish</param>
+    /// <param name="displayName">Display name of the task shown in the UI when creating pipeline run</param>
+    /// <returns>A new instance of <see cref="PublishTask"/> with the specified parameters</returns>
     /// </summary>
     protected static PublishTask Publish(string artifactName, string filePath, string? displayName = null)
```
Comment (Meir017) [inline src/Sharpliner/AzureDevOps/AzureDevOpsDefinition.cs:303]: I think we should consider having a `Publish` builder property here with two paths, one for `ArtifactType.Pipeline` and another for `ArtifactType.Filepath`. the property `FileSharePath` is required when `ArtifactType.Filepath`

Comment (premun) [reply src/Sharpliner/AzureDevOps/AzureDevOpsDefinition.cs:303]: Should we merge this and I will open an issue?

Comment (Meir017) [reply src/Sharpliner/AzureDevOps/AzureDevOpsDefinition.cs:303]: Yes

**Review by premun — APPROVED** [review]
Thanks!

**Pull request merged**